data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkService.java

```diff
@@ -106,7 +106,9 @@ void output(Collection<Record<Event>> records) {
                 final byte[] encodedBytes = encodedEvent.getBytes();
 
                 currentBuffer.writeEvent(encodedBytes);
-                bufferedEventHandles.add(event.getEventHandle());
```

Comment:
Should we add a unit test to confirm this is fixed? And help prevent future NPEs around this leak into this plugin?

Reply:
I actually tried this, but I was not able to get it to crash such that the test would actually show any verification.

Reply:
Null event handle would be the default case when not using end-to-end acknowledgements.

Reply:
@cmanning09 , I found that there was an unnecessary try-catch on `NullPointerException`. I removed that and was able to add a valid test.